module/x/gravity/keeper/ethereum_event_handler.go

```diff
@@ -100,40 +100,39 @@ func (a EthereumEventProcessor) verifyERC20DeployedEvent(ctx sdk.Context, event
 	// NOTE: This path is not encouraged and all supported assets should have
 	// metadata defined. If metadata cannot be defined, consider adding the token's
 	// metadata on the fly.
-	metadata := a.keeper.bankKeeper.GetDenomMetaData(ctx, event.CosmosDenom)
-	if metadata.Base == "" {
-		if event.Erc20Name != event.CosmosDenom {
-			return sdkerrors.Wrapf(
-				types.ErrInvalidERC20Event,
-				"invalid ERC20 name for token without metadata; got: %s, expected: %s", event.Erc20Name, event.CosmosDenom,
-			)
-		}
+	if metadata := a.keeper.bankKeeper.GetDenomMetaData(ctx, event.CosmosDenom); metadata.Base != "" {
+		return verifyERC20Token(metadata, event)
+	}
 
-		if event.Erc20Symbol != "" {
-			return sdkerrors.Wrapf(
-				types.ErrInvalidERC20Event,
-				"expected empty ERC20 symbol for token without metadata; got: %s", event.Erc20Symbol,
-			)
-		}
+	if event.Erc20Name != event.CosmosDenom {
+		return sdkerrors.Wrapf(
+			types.ErrInvalidERC20Event,
+			"invalid ERC20 name for token without metadata; got: %s, expected: %s", event.Erc20Name, event.CosmosDenom,
+		)
+	}
 
-		if event.Erc20Decimals != 0 {
-			return sdkerrors.Wrapf(
-				types.ErrInvalidERC20Event,
-				"expected zero ERC20 decimals for token without metadata; got: %d", event.Erc20Decimals,
-			)
-		}
+	if event.Erc20Symbol != "" {
+		return sdkerrors.Wrapf(
+			types.ErrInvalidERC20Event,
+			"expected empty ERC20 symbol for token without metadata; got: %s", event.Erc20Symbol,
+		)
+	}
 
-		return nil
+	if event.Erc20Decimals != 0 {
+		return sdkerrors.Wrapf(
+			types.ErrInvalidERC20Event,
+			"expected zero ERC20 decimals for token without metadata; got: %d", event.Erc20Decimals,
+		)
 	}
 
-	return verifyERC20Token(metadata, event)
+	return nil
 }
 
 func verifyERC20Token(metadata banktypes.Metadata, event *types.ERC20DeployedEvent) error {
 	if event.Erc20Name != metadata.Display {
 		return sdkerrors.Wrapf(
 			types.ErrInvalidERC20Event,
-			"ERC20 name %s does not match the denom display %s", event.Erc20Name, metadata.Description,
+			"ERC20 name %s does not match the denom display %s", event.Erc20Name, metadata.Display,
```

Comment:
@alexanderbez pretty sure this was a minor bug; mind confirming or denying? :)

Reply:
Yup. Good catch!